vivarium/core/engine.py

```diff
@@ -541,7 +716,14 @@ def apply_update(
         if process_updates:
             for path, process in process_updates:
                 assoc_path(self.processes, path, process)
-                self._add_process_path(process, path)
+                self._add_process_path(process, path, {})
+
+        if step_updates:
+            for path, step in step_updates:
+                dependencies = flow_update_dict.get(path)
+                assoc_path(self.steps, path, step)
+                self._add_step_path(step, path, dependencies)
```

Comment:
Is this tested anywhere? It would be good to test that division is applied correctly with `step_updates`.

Reply:
I added a test and fixed the issues that the test found